src/hotspot/share/runtime/synchronizer.cpp

```diff
@@ -1253,7 +1268,7 @@ static bool monitors_used_above_threshold(MonitorList* list) {
   if (NoAsyncDeflationProgressMax != 0 &&
       _no_progress_cnt >= NoAsyncDeflationProgressMax) {
     double remainder = (100.0 - MonitorUsedDeflationThreshold) / 100.0;
-    size_t new_ceiling = ceiling + (size_t)((double)ceiling * remainder) + 1;
+    size_t new_ceiling = ceiling / remainder + 1;
```

Comment:
The old calculation made no sense if the goal was to after NoAsyncDeflationProgressMax deflation cycles with not deflation stop trying until we get more monitors. With the current calculation we can keep running these no progress cycles for quite a number of iterations.

The remainder is a value in the range [0,1].  So it will always grow by at least 1. (When remainder == 1: `new_ceiling = ceiling + 1;`)

I do however realise that the new calculation needs to handle remainder == 0.0 (MonitorUsedDeflationThreshold = 100). But the whole NoAsyncDeflationProgressMax makes little sense if MonitorUsedDeflationThreshold is 0, as the value for new ceiling would only stop deflation if it is large enough that the `size_t monitor_usage = (monitors_used * 100LL) / ceiling;` gets truncated to 0. (Less than 1%). 

But this is a leftover from the deflation changes so I will probably just remove this change and create a RFE for reevaluating this.